app/src/main/java/ch/epfl/sdp/appart/filter/FilterViewModel.java

```diff
@@ -1,39 +1,64 @@
 package ch.epfl.sdp.appart.filter;
 
+import android.util.Log;
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 import ch.epfl.sdp.appart.ad.Ad;
 import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.location.geocoding.GoogleGeocodingService;
+import ch.epfl.sdp.appart.location.place.locality.Locality;
+import ch.epfl.sdp.appart.location.place.locality.LocalityFactory;
+import ch.epfl.sdp.appart.location.place.locality.MalformedLocalityException;
 import ch.epfl.sdp.appart.login.LoginService;
 import ch.epfl.sdp.appart.scrolling.card.Card;
 import ch.epfl.sdp.appart.user.User;
 import dagger.hilt.android.lifecycle.HiltViewModel;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 import javax.inject.Inject;
 
 @HiltViewModel
 public class FilterViewModel extends ViewModel {
 
   private List<Card> cardsFilter;
   final DatabaseService database;
+  final GoogleGeocodingService geocodingService;
   private int min;
   private int max;
-  /*
-  private String city;
-  private int range;
-   */
+  private Locality city;
+  private float range;
+
+
 
   @Inject
-  public FilterViewModel(DatabaseService database) {
+  public FilterViewModel(DatabaseService database, GoogleGeocodingService geocodingService) {
     this.database = database;
+    this.geocodingService = geocodingService;
   }
 
   public CompletableFuture<Boolean> confirmFilter(){
       CompletableFuture<List<Card>> queriedCards = database.getCardsFilterPrice(min, max);
-      return queriedCards.thenApply(v -> {
-        cardsFilter = v;
+      return queriedCards.thenApply(cardList -> {
+        List<Card> cardsFilterBuffer = new ArrayList<>();
+        for (Card c : cardList){
+          //remove postal code
+          String [] l = c.getCity().split(" ", 2);
+          try {
+            Locality location = LocalityFactory.makeLocality(l[1]);
+            float dist;
+            dist = geocodingService.getDistance(location, city).get();
+            //conversion in kilometers
+            dist = dist/1000;
+            if(dist <= range){
+              cardsFilterBuffer.add(c);
+            }
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
```

Comment:
For reference in the main comment

Reply:
Yes, indeed there are many exception that are throws when you click the confirm button. This is due to the bad format of some announce (location that doesn't exist, ...) But with this general **catch (Exception e)** all this exception are ignored and so after the filter we display only the ad with the correct information. So, in theory when you make the filter you should see all the add filter and in the console there are the print of the error, but the application doesn't crash no ? (I try to run the app  local and its work for me). Let me know if it more clear and if there still is the issue.

Reply:
I checked again and now it indeed works! I don't know what was happening before